### ISSUE TITLE
Call to windows API causing FPS issues

### ISSUE THREAD

**Issue by mirdoll** [citizenfx/fivem#1082]
Hello!
So this have been an issue for the past 2 months. 
The issue involves consistent stutter and frame drops for various users on different servers.
I have ran profilers aswell as resmon to make sure it's not a server sided issue. I have also done a spec check on 2 different PC's getting the same issue. (You can find the documentation here: https://www.reddit.com/r/FiveM/comments/q8ocwx/stutter_and_lag_on_gpucpu_drop/ )

Upon trying to find a fix for this I've found several different threads regarding the same issue starting to surface around the same time, but no solution:
https://www.reddit.com/r/FiveM/comments/pmzbwh/a_call_into_the_windows_api_took_to_long_recently/
https://www.reddit.com/r/FiveM/comments/q4nbyf/error_issue_causing_fps_issues/
https://www.reddit.com/r/FiveM/comments/pn1zvh/sudden_stuttering_when_gpu_usage_drops/

**Comment by mirdoll** [comment]
Thank you for the response!
I've seen some complaints regarding the same thing on the forums aswell (all within 2 months: https://forum.cfx.re/search?q=stutter%20order%3Alatest), but regardless I've forwarded this issue into a topic on the cfx forums to hope it will get looked into. 
I've also recorded a ETW log that I can forward in DM on there if requested.

**Comment by duck-37** [comment]
> Thank you for the response! I've seen some complaints regarding the same thing on the forums aswell (all within 2 months: https://forum.cfx.re/search?q=stutter%20order%3Alatest), but regardless I've forwarded this issue into a topic on the cfx forums to hope it will get looked into. I've also recorded a ETW log that I can forward in DM on there if requested.

Yeah, those forum topics should've been looked at.. sorry about that. Anwyays, if you could forward the trace to my [forum account](https://forum.cfx.re/u/_duk/summary) that would be great. I'll post updates in here when I can. Thanks!

**Comment by duck-37** [comment]
Quick update: Can you check to see whether you have this option enabled? 
![image](https://user-images.githubusercontent.com/74797529/137653447-2be66769-4731-4a2a-a689-2a07b9cbd911.png)

**Comment by renatolgarcia** [comment]
Hi! I have the same problem reported in this forum. I keep FiveM open and the task manager. during FPS drops, where the slowness alert appears due to Windows APIs, the computer is not overloaded.

My setup:

AMD Ryzen 5 3600
32GB RAM 3600Mhz
Zotak Gaming RTX 3080 12GB
SSD M.2 1tb
XPG 850W power supply
Asus TUF Gaming X570-plus motherboard

**Comment by blattersturm** [comment]
This should be fixed in newer Windows versions: https://blogs.windows.com/windowsdeveloper/2023/05/26/delivering-delightful-performance-for-more-than-one-billion-users-worldwide/ (see the heading of "Reduced game stutter with high report rate mice").

**Comment by popnatedogg** [comment]
is this still active i have the same problem for a while now and i even switched from a 1080 gtx to a 3060 rtx and nothing changed still getting major fps drops

**Comment by Pitottt** [comment]
I had this problem, so when I move my mouse this problem appears but when i don't move my mouse it don't happen. Apperantly high polling rate made this slow system call thing, i switched from 8k polling rate to 1k polling rate and the fps drop stops happening. Hope this can help you

**Comment by orochi85** [comment]
I discovered that my webcam was disconnecting and causing this error. Check USB devices

**Comment by chrishandsomeusss** [comment]
@duk-37 Did you ever find a fix for this? I built an almost 3000 dollar PC in the summer, and have been dealing with this issue ever since (4ish months). My old PC was 16gb of ram, 8gb graphics card, and my CPU was at like 95% when playing, but never once had this issue but i suspect it was because i had windows 10 as that pc could not upgrade. 

As of now the issue still happens, and always works like this like the original poster says which is frustrating - It happens anytime, any place, and never doing the same thing. Sometimes talking to 1 person, sometimes randomly at an event, sometimes while moving, and sometimes while standing still. 

When this issue happens, the CPU and GPU usage drop a lot, so all task manager stats are like CPU 5% and GPU 10% instead of normal CPU 15% and GPU 40%, and then the entire PC lags until it corrects. Even opening task manager, and anything on my other screen, is SUPER LAGGY as the whole PC slows down even with those above performance stats all being low while the issue is occurring. Also sometimes the issue might happen for 10 seconds and fix, or it might stay for 1-2-3 mins until it corrects, and the FPS goes up to normal and the PC becomes stable. 

Had this brand new PC built by a PC SHOP locally, and it is fully updated all around and i keep it and the drivers fully updated. Specs are also:
GPU = RX 7600 XT
CPU = ryzen 9 9900x
RAM = 32gb DDR5 (EXPO Turned ON so ram speed = 6000MT/s)
(Issue occurs with expo on or off I have tested both over a week each)
2TB Netac NVMe SSD PCIe
1TB Regular Standard SSD (I can dig out the box if needed)
[Tried GTA v and Fivem reinstalled on both SSD's with same issue on both]


Again when the issue happens yes the pop up comes up at the bottom "A call into the Windows API took too long recently and led to a game stutter". I am coming here as I have searched far and wide for someone like you who knows the deep smart computer stuff that no one really does on any of these forums.
